src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Weather/Manager.cpp

```diff
@@ -0,0 +1,790 @@
+        CSSISpaceWeatherIndex_ = 0;
+
+        for (const auto& CSSISpaceWeather : CSSISpaceWeatherArray_)
+        {
+            if (CSSISpaceWeather.accessDailyPredictionInterval().contains(anInstant))
+            {
+                return &CSSISpaceWeather;
+            }
+
+            CSSISpaceWeatherIndex_++;
+        }
+    }
+
+    // Try monthly prediction span of loaded space weather files
+
+    {
+        CSSISpaceWeatherIndex_ = 0;
+
+        for (const auto& CSSISpaceWeather : CSSISpaceWeatherArray_)
+        {
+            if (CSSISpaceWeather.accessMonthlyPredictionInterval().contains(anInstant))
+            {
+                return &CSSISpaceWeather;
+            }
+
+            CSSISpaceWeatherIndex_++;
+        }
+    }
+
+    // No space weather data found
+
+    {
+        CSSISpaceWeatherIndex_ = 0;
+
+        return nullptr;
+    }
+}
+
+File Manager::getLocalRepositoryLockFile() const
+{
+    using ostk::core::fs::Path;
+
+    return File::Path(localRepository_.getPath() + Path::Parse(".lock"));
+}
+
+File Manager::getLatestCSSISpaceWeatherFile() const
+{
+    // Parse CSSI Space Weather Directories, e.g.,
+    // `.open-space-toolkit/physics/environment/atmospheric/earth/weather/CSSI-Space-Weather/2022-05-19/`, and find the
+    // latest one.
+
+    Map<Instant, File> CSSISpaceWeatherMap = {};
+
+    for (const auto& directory : this->getCSSISpaceWeatherDirectory().getDirectories())
+    {
+        if ((directory.getName() != temporaryDirectoryName) && directory.containsFileWithName(CSSISpaceWeatherFileName))
+        {
+            const Date date = Date::Parse(directory.getName());
+
+            const Instant instant = Instant::DateTime({date, Time::Midnight()}, Scale::UTC);
+
+            const File bulletinFile = File::Path(directory.getPath() + Path::Parse(CSSISpaceWeatherFileName));
+
+            CSSISpaceWeatherMap.insert({instant, bulletinFile});
+        }
+    }
+
+    if (!CSSISpaceWeatherMap.empty())
+    {
+        return CSSISpaceWeatherMap.rbegin()->second;
+    }
+
+    return const_cast<Manager*>(this)->fetchLatestCSSISpaceWeather_();
+}
+
+void Manager::setup()
+{
+    using ostk::core::fs::Path;
+    using ostk::core::fs::File;
+
+    if (!localRepository_.exists())
+    {
+        localRepository_.create();
+    }
+
+    if (!this->getCSSISpaceWeatherDirectory().exists())
+    {
+        this->getCSSISpaceWeatherDirectory().create();
+    }
+}
+
+void Manager::loadCSSISpaceWeather_(const CSSISpaceWeather& aCSSISpaceWeather)
+{
+    for (const auto& CSSISpaceWeather : CSSISpaceWeatherArray_)
+    {
+        if (CSSISpaceWeather.accessLastObservationDate() == aCSSISpaceWeather.accessLastObservationDate())
+        {
+            throw ostk::core::error::RuntimeError("CSSI Space Weather already added.");
+        }
+    }
+
+    CSSISpaceWeatherArray_.add(aCSSISpaceWeather);  // [TBI] Add in ascending time order
+
+    CSSISpaceWeatherIndex_ = 0;
+}
+
+File Manager::fetchLatestCSSISpaceWeather_()
```

Comment:
Why underscore in function name?

Reply:
These functions are private in the header

Reply:
Interesting gotcha. I didn't realize we use an underscore in function naming as well. Let me take a look.

Reply:
Ah gotcha it's cause there is another function with the same name, so we have a private one with an _. Gotcha.